### PR TITLE
Vita snake

### PR DESCRIPTION
**Pull request by slobters**
# New Port for { vitaSnake }

## Game Information
- **Title**: { vitaSnake }
- **URL**: { https://github.com/Grzybojad/vitaSnake }

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [ X] AmberELEC
- [X ] ArkOS
- [ X] ROCKNIX
- [ X] muOS

### Resolution Tests
Test all major resolutions:
- [X ] 480x320
- [X ] 640x480
- [X ] 720x720 (RGB30)
- [X ] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/vitasnake/cover.png
Comment (cdeletre) [inline ports/vitasnake/cover.png:1]: it's possible to optimize the png size with pngquant (down to 56 kB) or online with https://squoosh.app/

File: ports/vitasnake/log.txt
Comment (cdeletre) [inline ports/vitasnake/log.txt:1]: This file can be removed

File: ports/vitasnake/screenshot.png
Comment (cdeletre) [inline ports/vitasnake/screenshot.png:1]: image weight can be reduced to 92 kB with pngquant. You can also use https://squoosh.app/

File: ports/vitasnake/splash.png
Comment (cdeletre) [inline ports/vitasnake/splash.png:1]: It's a duplicate of `gamedata/splash.png` so maybe it can be removed ?

File: ports/vitasnake/vitaSnake.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+GAMEDIR="/$directory/ports/vitasnake"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$LD_LIBRARY_PATH"
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/gamedata/"
+export GMLOADER_PLATFORM="os_linux"
+
+# We log the execution of the script into log.txt
+exec > >(tee "$GAMEDIR/log.txt") 2>&1
```
Comment (cdeletre) [inline ports/vitasnake/vitaSnake.sh:32]: Old logging method we now use a newer version which overwrites itself

`"$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1`

File: ports/vitasnake/vitasnake.md
Comment (cdeletre) [inline ports/vitasnake/vitasnake.md:1]: This file should be renamed as `README.md`

**Comment by cdeletre** [comment]
The port file/folder structure must be revised, it should be something like this:
```
spookynakki % tree
.
├── README.md
├── Vitasnake.sh
├── cover.png
├── gameinfo.xml
├── port.json
├── screenshot.png
└── vitasnake
    ├── game.apk
    ├── gamedata
    ├── gmloader
    └── libs
```

**Review by cdeletre — CHANGES_REQUESTED** [review]
The port file/folder structure must be revised, it should be something like this:
```
Portmaster-New
└──ports
    └──vitasnake
        ├── README.md
        ├── Vitasnake.sh
        ├── cover.png
        ├── gameinfo.xml
        ├── port.json
        ├── screenshot.png
        └── vitasnake
            ├── game.apk
            ├── gamedata
            ├── gmloader
            ├── ...
            └── libs
```

Also `port.json` is missing

File: ports/vitasnake/port.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "genres": [
+      "action",
+      "arcade"
+    ],
+    "image": null,
+    "rtr": true,
+    "exp": false,
+    "runtime": null,
+    "reqs": [
+      "power"
```
Comment (Cebion) [inline ports/vitasnake/port.json:27]: This requirement is set if it only works on cpus above RK3326 is this correct?

**Comment by slobters** [comment]
I wasn't sure how to test this so I set it like that.
But this game should be able to run on something like RK3326.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Review by JanTrueno — APPROVED** [review]

File: ports/vitasnake/cover.png
Comment (JanTrueno) [inline ports/vitasnake/cover.png:1]: Cover needs to be 640x480.

**Review by JanTrueno — CHANGES_REQUESTED** [review]

File: ports/vitasnake/cover.png
Comment (slobters) [reply ports/vitasnake/cover.png:1]: Cover is 640x480 from 16:9 image with black bars.
Shoul I stretch or croop original image to 640x480?

Comment (slobters) [reply ports/vitasnake/cover.png:1]: Oh wait cover, it's always 400x300 in the ports when I download them.
Screenshot is fine and cover need to be 640x480?
Why covers have 400x300 inside ports then?
Covers get shrinked for muOS?

Comment (JanTrueno) [reply ports/vitasnake/cover.png:1]: Not sure which ports you are talking about, but both screenshot and cover have to be 640x480 :)

Comment (JanTrueno) [reply ports/vitasnake/cover.png:1]: Maybe they do get shrunk for MuOS, thats very possible. I dont know though.

Comment (slobters) [reply ports/vitasnake/cover.png:1]: I will make new cover later today.
Thanks for help.

**Review by JanTrueno — APPROVED** [review]

**Review by Cebion — CHANGES_REQUESTED** [review]
The file structure is wrong.
In the top vitasnake folder you only need readme, cover, gameinfo.xml, portj.json, screenshot and the .sh

**Comment by slobters** [comment]
Fixed!
I moved the licence file into the second vitasnake folder. Is that ok?

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Review by Cebion — APPROVED** [review]

**Review by JanTrueno — APPROVED** [review]

**Pull request merged**